### PR TITLE
Fix unexpected closing of signal connection during reconnect

### PR DESCRIPTION
**Pull request by lukasIO**

### CODE REVIEW

File: src/api/SignalClient.ts [hunk 1]
```diff
@@ -350,6 +360,8 @@ export class SignalClient {
                 resolve();
                 shouldProcessMessage = true;
               }
+            } else if (this.isEstablishingConnection && resp.message.case === 'leave') {
+              reject(new ConnectionError('Received leave request while trying to (re)connect'));
```
Comment (lukasIO) [inline src/api/SignalClient.ts:364]: not sure if this is the intended behaviour: 

This means if a user is reconnecting and the very first message they receive is not a `ReconnectResponse` but rather a leave request, we mark the reconnection attempt as failed, which will result in a new reconnection attempt happening.

Comment (boks1971) [reply src/api/SignalClient.ts:364]: The intended behaviour there is for the client to do a full reconnect. Server sends `LeaveRequest` when it cannot resume. So, a full reconnect is needed. Is the new reconnection attempt a full reconnect?

**Review by boks1971 — APPROVED** [review]
LGTM, but please get @davidzhao 's 👀 as I am not aware of all the subtleties here.

**Review by davidzhao — APPROVED** [review]
lgtm!

**Pull request merged**